golang/cosmos/x/vstorage/keeper/keeper.go

```diff
@@ -61,6 +130,29 @@ func (k Keeper) ImportStorage(ctx sdk.Context, entries []*types.DataEntry) {
 	}
 }
 
+func (k Keeper) EmitChange(ctx sdk.Context, change *Change) {
+	if change.Value == change.OldValue {
```

Comment:
Consider detecting this in `ChangeManager.Track()` and removing the cache entry instead. This would avoid the overhead of storing and sorting no-op changes, in case those are common.

Reply:
Unfortunately, I can't without changing the semantics, because the change stores `isLegacy` which needs to be tracked until the event is emitted.